lib/control-connection.js

```diff
@@ -95,7 +95,7 @@ ControlConnection.prototype.init = function (callback) {
           var portSeparatorIndex = name.lastIndexOf(']:');
           return addHost(name.substr(1, portSeparatorIndex - 1), name.substr(portSeparatorIndex + 2), eachNext);
         }
-        if (name.indexOf('.') > 0 && name.indexOf(':') > 0) {
+        if (name.indexOf(':') > 0) {
           // IPv4 with port notation
           var parts = name.split(':');
           return addHost(parts[0], parts[1], eachNext);
```

Comment:
To avoid breaking ipv6 support, we should add a condition that parts is equals to 2.
ie: 
- `::1` -> ipv6
- `[::1]:9042` -> ipv6 with port already covered by previous condition.
- `whatever:9042` -> ipv4 or host with port